devicetypes/smartthings/zwave-panic-button.src/zwave-panic-button.groovy

```diff
@@ -0,0 +1,149 @@
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		main(["panic"])
+
+		details(["panic", "buttons", "battery"])
+	}
+}
+
+def installed() {
+	log.debug "Installed $device.displayName"
+	sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "zwave", scheme:"untracked"].encodeAsJson(), displayed: false)
+	if(hasMultipleButtons()) {
+		addChildButtons()
+	}
+	setPanicAlarm("clear")
+	def cmds = [
+			zwave.batteryV1.batteryGet().format(),
+			zwave.notificationV3.notificationGet().format(),
+			zwave.wakeUpV1.wakeUpNoMoreInformation().format()
+	]
+	response(cmds)
+}
+
+def parse(String description) {
+	def result
+	def cmd = zwave.parse(description)
+
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse ${description} to ${result}"
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def button = cmd.sceneNumber
+	def state = (cmd.keyAttributes == 3) ? "double" : "pushed"
+	setButtonState(button, state)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def status = cmd.event ? "panic" : "clear"
+	setPanicAlarm(status)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	def status = cmd.sensorValue ? "panic" : "clear"
+	setPanicAlarm(status)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def results = [createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)]
+
+	def prevBattery = device.currentState("battery")
+	if (!prevBattery || (new Date().time - prevBattery.date.time)/60000 >= 60 * 53) {
+		results += response(zwave.batteryV1.batteryGet().format())
+	}
+	results += response(zwave.wakeUpV1.wakeUpNoMoreInformation().format())
+	results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [ name: "battery", unit: "%" ]
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "${device.displayName} has a low battery"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled zwave command $cmd"
+	[:]
+}
+
+private setPanicAlarm(status) {
+	createEvent(name: "panicAlarm", value: status, descriptionText: "Panic Alarm status: ${status}", displayed: true, isStateChange: true)
+}
+
+private setButtonState(buttonId, state) {
+	def event
+	String childDni = "${device.deviceNetworkId}/$buttonId"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	if (!child) {
+		log.error "Child device $childDni not found"
+	} else {
+		child?.sendEvent(name: "button", value: "$state")
+		event = createEvent(descriptionText: "Button $buttonId was $state", isStateChange: true)
+	}
+	event
+}
+
+private addChildButtons() {
+	log.debug "Creating child devices"
+	for (i in 1..3) {
+		String childDni = "${device.deviceNetworkId}/$i"
```

Comment:
We've been trying to be a bit more consistent with how we handle child DNIs and on the zigbee side we have settled on `<deviceDNI>:<endpoint>` so using the colon as a separator. Could you change this to be the same so we can try to have consistency across devices.